### PR TITLE
Add a dockerExcludes override parameter

### PR DESCRIPTION
**Pull request by M-Davies**
Adds a parameter that overrides our current system of always building docker if there is dockerImage and dockerFile parameter.

Closes: #1997 
Signed-off-by: Morgan Davies <morgan.davies@ibm.com>

### CODE REVIEW

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
Had to use a `stringParam` instead of a `textParam` for this because `textParam` doesn't allow you to specify a `defaultValue` as far as I can see

File: pipelines/build/common/build_base_file.groovy [hunk 1]
```diff
@@ -132,27 +133,56 @@ class Builder implements Serializable {
...
+        Boolean overrideDocker = false
+        if (dockerExcludes == {}) {
+            return overrideDocker 
+        }
+
+        String stringArch = configuration.arch as String
+        String stringOs = configuration.os as String
+        String estimatedKey = stringArch + stringOs.capitalize()
+
+        if (configuration.containsKey("additionalFileNameTag")) {
```
Comment (karianna) [inline pipelines/build/common/build_base_file.groovy:146]: Possibly brittle?  Are there other suffixes that might apply when additionalFileNameTag is detected?

Comment (M-Davies) [reply pipelines/build/common/build_base_file.groovy:146]: Not that I know of. This code is also used to [form node labels](https://github.com/AdoptOpenJDK/openjdk-build/blob/master/pipelines/build/common/build_base_file.groovy#L176) that helps jenkins pick which machines should run which builds so if this is brittle, that makes the linked code brittle too

**Pull request merged**